### ISSUE TITLE
Calling a deprecated version of the Ads API

### ISSUE THREAD

**Issue by ales-cervenka** [singer-io/tap-facebook#49]
When I ran the tap-facebook today (Sept 13th, 2018), I got the following error:

```
CRITICAL 
CRITICAL 
CRITICAL   Message: Call was not successful
CRITICAL   Method:  GET
CRITICAL   Path:    https://graph.facebook.com/v3.0/me/adaccounts
CRITICAL   Params:  {'summary': 'true'}
CRITICAL 
CRITICAL   Status:  400
CRITICAL   Response:
CRITICAL     {
CRITICAL       "error": {
CRITICAL         "message": "(#2635) You are calling a deprecated version of the Ads API. Please update to the latest version: v3.1.",
CRITICAL         "type": "OAuthException",
CRITICAL         "code": 2635,
CRITICAL         "fbtrace_id": "DLTgDI8GfSs"
CRITICAL       }
CRITICAL     }
Traceback (most recent call last):
  File "/root/venv/bin/tap-facebook", line 11, in <module>
    sys.exit(main())
  File "/root/venv/lib/python3.6/site-packages/tap_facebook/__init__.py", line 660, in main
    raise e
  File "/root/venv/lib/python3.6/site-packages/tap_facebook/__init__.py", line 652, in main
    main_impl()
  File "/root/venv/lib/python3.6/site-packages/tap_facebook/__init__.py", line 633, in main_impl
    accounts = user.get_ad_accounts()
  File "/root/venv/lib/python3.6/site-packages/facebook_business/adobjects/user.py", line 234, in get_ad_accounts
    return request.execute()
  File "/root/venv/lib/python3.6/site-packages/facebook_business/api.py", line 657, in execute
    cursor.load_next_page()
  File "/root/venv/lib/python3.6/site-packages/facebook_business/api.py", line 801, in load_next_page
    params=self.params,
  File "/root/venv/lib/python3.6/site-packages/facebook_business/api.py", line 330, in call
    raise fb_response.error()
facebook_business.exceptions.FacebookRequestError: 

  Message: Call was not successful
  Method:  GET
  Path:    https://graph.facebook.com/v3.0/me/adaccounts
  Params:  {'summary': 'true'}

  Status:  400
  Response:
    {
      "error": {
        "message": "(#2635) You are calling a deprecated version of the Ads API. Please update to the latest version: v3.1.",
        "type": "OAuthException",
        "code": 2635,
        "fbtrace_id": "DLTgDI8GfSs"
      }
    }
```

Any thoughts? 

By the way, I have never run this tap before.

Thank you,
Ales

**Comment by KAllan357** [comment]
@ales-cervenka Thanks for the report. I'm surprised this isn't working. Since you are running this locally, what you can try is updating the `setup.py` file to use a newer version of the Facebook Python SDK. https://github.com/singer-io/tap-facebook/blob/master/setup.py#L15

Try running the tap with version 3.1.4. That would end up using version v3.1 of the Facebook Ads APIs.

**Comment by ales-cervenka** [comment]
@KAllan357 thanks for the reply. I wish I knew how to fix the version of the facebook_business dependency. I installed the tap-facebook using virtualenv, as recommended:

virtualenv -p python3 venv
source venv/bin/activate
pip install tap-facebook

I am not able to locate any setup.py  file in the venv directory.

**Comment by mk556** [comment]
how to run the tap after changing the version to `3.1.4` in `setup.py`?

**Comment by cmerrick** [comment]
Fixed by #59
